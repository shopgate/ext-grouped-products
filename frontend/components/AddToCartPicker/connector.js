import { connect } from 'react-redux';
import addProductsToCart from '@shopgate/pwa-common-commerce/cart/actions/addProductsToCart';

/**
 * Connects the dispatch function to a callable function in the props.
 * @param {Function} dispatch The redux dispatch function.
 * @param {Object} props The component props.
 * @return {Object} The extended component props.
 */
const mapDispatchToProps = (dispatch, { productId }) => ({
  addProductToCart: (quantity) => {
    dispatch(addProductsToCart([{
      productId,
      quantity,
    }]));
  },
});

export default connect(null, mapDispatchToProps);
