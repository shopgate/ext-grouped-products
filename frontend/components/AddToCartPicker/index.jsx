import React, { Component } from 'react';
import PropTypes from 'prop-types';
import isEqual from 'lodash/isEqual';
import BasePicker from '@shopgate/pwa-common/components/Picker';
import Sheet from '@shopgate/pwa-ui-shared/Sheet';
import List from './components/List';
import AddToCartButton from './components/AddToCartButton';
import { ADD_TO_CART_BUTTON_TYPE_DEFAULT } from '../../constants';
import { createPickerItems } from '../../helpers';
import styles from './style';

/**
 * The AddToCartPicker component.
 */
class AddToCartPicker extends Component {
  static propTypes = {
    conditioner: PropTypes.shape().isRequired,
    handleAddToCart: PropTypes.func.isRequired,
    buttonProps: PropTypes.shape(),
    clickDelay: PropTypes.number,
    productId: PropTypes.string,
    stock: PropTypes.shape({
      ignoreQuantity: PropTypes.bool,
      quantity: PropTypes.number,
      orderable: PropTypes.bool,
    }),
  };

  static defaultProps = {
    /**
     * Time in ms that delays picker interaction in order
     * to let animations complete first.
     */
    buttonProps: {
      handleAddToCart: /* istanbul ignore next */ () => {},
      onClick: () => {},
      isDisabled: false,
      isLoading: false,
      hasLoading: true,
      noShadow: false,
      type: ADD_TO_CART_BUTTON_TYPE_DEFAULT,
    },
    clickDelay: 150,
    productId: '',
    stock: null,
  };

  static contextTypes = {
    i18n: PropTypes.func,
  };

  /**
   * Constructor
   * @param {Object} props The component props.
   * @param {Object} context The component context.
   */
  constructor(props, context) {
    super(props, context);

    // Initialize the added quantity. The state is used to trigger the AddToCartButton animation.
    this.state = {
      addedQuantity: 0,
    };

    const { __ } = context.i18n();

    // Prepare a modal component for the BasePicker.
    this.modalComponent = modalProps =>
      (<Sheet {...modalProps} title={__('grouped_products.add_to_cart_picker.headline')} />);

    // Prepare a list component for the BasePicker
    this.listComponent = ({
      items, onSelect,
    }) => (
      <List>
        {items.map(item => (
          <List.Item
            key={item.value}
            title={item.label}
            onClick={() => {
                setTimeout(() => {
                  onSelect(item.value);
                }, this.props.clickDelay);
              }}
          />
          ))}
      </List>
    );
  }

  /**
   * Only update when the `addedQuantity` state changes. To avoid that the
   * picker closes when the Redux state changes.
   * @param {Object} nextProps The next component props.
   * @param {Object} nextState The next component state.
   * @return {boolean}
   */
  shouldComponentUpdate(nextProps, nextState) {
    return (nextState.addedQuantity !== this.state.addedQuantity) ||
      !isEqual(nextProps.buttonProps, this.props.buttonProps);
  }

  /**
   * Returns the props for the picker button.
   * @return {Object}
   */
  get buttonProps() {
    let props = {
      ...this.constructor.defaultProps.buttonProps,
      ...this.props.buttonProps,
      conditioner: this.props.conditioner,
      addedQuantity: this.state.addedQuantity,
      className: null,
    };

    if (props.noShadow) {
      // Inject additional classes when the button is rendered without a shadow.
      props = {
        ...props,
        className: props.isDisabled ? styles.buttonFlatDisabled : styles.buttonFlat,
      };
    }

    return props;
  }

  /**
   * Puts a product into the cart.
   * @param {number} quantity The desired quantity.
   */
  handleAddToCart = (quantity) => {
    this.props.handleAddToCart(quantity);

    this.setState({
      addedQuantity: this.state.addedQuantity + quantity,
    });
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    const pickerItems = createPickerItems(this.props.stock);
    return (
      <BasePicker
        onClick={this.handleAddToCart}
        items={pickerItems}
        className={styles.buttonContainer}
        modalComponent={this.modalComponent}
        buttonProps={this.buttonProps}
        buttonComponent={AddToCartButton}
        listComponent={this.listComponent}
        onSelect={this.handleAddToCart}
      />
    );
  }
}

export default AddToCartPicker;
