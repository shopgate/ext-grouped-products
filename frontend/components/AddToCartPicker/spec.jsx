import React, { Component } from 'react';
import { createWrappedComponent } from '../mockStore';
import {
  mockedState,
  mockedProduct,
  mockedIgnoredQuantityProduct,
  mockedNotOrderableProduct,
} from '../mock';

// eslint-disable-next-line require-jsdoc
class mockedAdToCartButton extends Component {
  // eslint-disable-next-line require-jsdoc
  static get propTypes() {
    return {};
  }

  // eslint-disable-next-line require-jsdoc
  constructor(props) {
    super(props);
    this.state = {
      // eslint-disable-next-line react/no-unused-state
      showCheckMark: false,
    };
  }

  // eslint-disable-next-line require-jsdoc
  render() {
    return <div><button onClick={this.handleClick}>Add to cart</button></div>;
  }
}
// Missing css-spring mock on pwa-ui-shared
jest.mock('@shopgate/pwa-ui-shared/AddToCartButton', () => mockedAdToCartButton);

jest.mock('@shopgate/pwa-ui-shared/AddToCartButton/style', () => (
  {
    buttonSize: 10,
    iconSize: 10,
    buttonWrapper: () => { },
    buttonWrapperNoShadow: () => { },
  }
));

jest.mock('../config', () => { }, { virtual: true });

jest.mock('@shopgate/pwa-common-commerce/cart/actions/addProductsToCart', () =>
  jest.fn().mockReturnValue('mocked_add_products_to_cart_action'));

jest.mock('../../config', () => ({
  maxQuantityPickerEntries: 5,
}), { virtual: true });

jest.mock('react-portal', () => (
  ({ isOpened, children }) => (
    isOpened ? children : null
  )
));
try {
  jest.doMock('@shopgate/pwa-common/node_modules/react-portal', () => (
    ({ isOpened, children }) => (
      isOpened ? children : null
    )
  ));
} catch (e) {
  // Do nothing.
}

jest.mock('@shopgate/pwa-ui-shared/Sheet', () => ({ children }) => children);

const mockHandleAddToCart = jest.fn();

jest.useFakeTimers();

describe('<AddToCartPicker />', () => {
  // eslint-disable-next-line global-require
  const AddToCartPicker = require('./index.jsx').default;

  /**
   * Creates a component with a provided store state.
   * @param {Object} product A product to set the component props for.
   * @param {Object} [buttonProps={}] Mocked button props.
   * @return {ReactWrapper}
   */
  const createComponent = (product, buttonProps = {}) => {
    const { id, stock } = product;

    const mockedProps = {
      buttonProps,
      productId: id,
      handleAddToCart: mockHandleAddToCart,
      stock,
      onClose: () => { },
    };

    return createWrappedComponent(AddToCartPicker, mockedState, mockedProps);
  };

  it('should create picker items like expected when quantity is not ignored', () => {
    const component = createComponent(mockedProduct);
    const picker = component.find('Picker').first();
    const pickerItems = picker.prop('items');
    expect(pickerItems).toHaveLength(4);
    expect(pickerItems[0]).toEqual({
      label: '1',
      value: 1,
    });
    expect(pickerItems[pickerItems.length - 1]).toEqual({
      label: '4',
      value: 4,
    });
  });

  it('should create picker items like expected when quantity is ignored', () => {
    const component = createComponent(mockedIgnoredQuantityProduct);
    const picker = component.find('Picker').first();
    const pickerItems = picker.prop('items');
    expect(pickerItems).toHaveLength(5);
    expect(pickerItems[0]).toEqual({
      label: '1',
      value: 1,
    });
    expect(pickerItems[pickerItems.length - 1]).toEqual({
      label: '5',
      value: 5,
    });
  });

  it('should update the component when expected', () => {
    const component = createComponent(mockedProduct);
    const picker = component.find('AddToCartPicker').first();

    const props = picker.props();
    const state = { ...picker.instance().state };

    const scu = picker.instance().shouldComponentUpdate.bind(picker.instance());

    // Nothing changed
    expect(scu({ ...props }, { ...state })).toBe(false);

    // Some ignored prop changed
    expect(scu({
      ...props,
      clickDelay: 200,
    }, { ...state })).toBe(false);

    // Button props changed
    expect(scu({
      ...props,
      buttonProps: {
        ...props.buttonProps,
        isLoading: true,
      },
    }, { ...state })).toBe(true);

    // State changed
    expect(scu({
      ...props,
    }, {
      ...state,
      addedQuantity: 3,
    })).toBe(true);
  });
});
