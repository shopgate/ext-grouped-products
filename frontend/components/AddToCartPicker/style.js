import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { variables, colors } = themeConfig;

const buttonContainer = css({
  marginBottom: variables.gap.xsmall,
}).toString();

const buttonFlat = css({
  boxShadow: `inset 0 0 0 2px ${colors.primary}`,
}).toString();

const buttonFlatDisabled = css({
  // Empty class to overwrite the default styling of the button.
}).toString();

export default {
  buttonContainer,
  buttonFlat,
  buttonFlatDisabled,
};
