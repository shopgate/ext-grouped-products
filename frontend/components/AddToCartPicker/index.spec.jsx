import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import addProductsToCart from '@shopgate/pwa-common-commerce/cart/actions/addProductsToCart';
import {
  mockedProduct,
  mockedLowQuantityProduct,
} from '../GroupedProducts/mock';

import AddToCartPicker from './index';

jest.mock('@shopgate/pwa-common-commerce/cart/actions/addProductsToCart', () =>
  jest.fn().mockReturnValue('mocked_add_products_to_cart_action'));

jest.mock('../../config', () => ({
  maxQuantityPickerEntries: 5,
}));

jest.mock('react-portal', () => (
  ({ isOpened, children }) => (
    isOpened ? children : null
  )
));
try {
  jest.doMock('@shopgate/pwa-common/node_modules/react-portal', () => (
    ({ isOpened, children }) => (
      isOpened ? children : null
    )
  ));
} catch (e) {
  // Do nothing.
}

jest.mock('@shopgate/pwa-ui-shared/Sheet', () => ({ children }) => children);

const mockedStore = configureStore();
const mockDispatch = jest.fn();

/**
 * Creates component with provided store state.
 * @param {Object} product A product to set the component props for.
 * @return {ReactWrapper}
 */
const createComponent = (product) => {
  const store = mockedStore({});
  store.dispatch = mockDispatch;

  const { id, stock } = product;
  return mount(
    <Provider store={store}>
      <AddToCartPicker productId={id} stock={stock} />
    </Provider>,
    mockRenderOptions
  );
};

jest.useFakeTimers();

describe('<AddToCartPicker />', () => {
  it('should render the component as expected', () => {
    const { id, stock } = mockedProduct;
    const component = createComponent(mockedProduct);
    component.simulate('click');
    expect(component).toMatchSnapshot();

    expect(component.find(AddToCartPicker).first().prop('productId')).toEqual(id);
    expect(component.find(AddToCartPicker).first().prop('stock')).toEqual(stock);

    const pickerEntry = component.find('ListItem Item').first();

    pickerEntry.simulate('click');
    jest.runAllTimers();

    expect(mockDispatch).toHaveBeenCalledTimes(1);
    expect(mockDispatch).toHaveBeenCalledWith(addProductsToCart());
  });

  it('should create picker items like expected when quantity is not ignored', () => {
    const component = createComponent(mockedProduct);
    const picker = component.find('Picker').first();
    const pickerItems = picker.prop('items');
    expect(pickerItems).toHaveLength(4);
    expect(pickerItems[0]).toEqual({
      label: '1',
      value: 1,
    });
    expect(pickerItems[pickerItems.length - 1]).toEqual({
      label: '4',
      value: 4,
    });
  });

  it('should create picker items like expected when quantity is ignored', () => {
    const component = createComponent(mockedLowQuantityProduct);
    const picker = component.find('Picker').first();
    const pickerItems = picker.prop('items');
    expect(pickerItems).toHaveLength(5);
    expect(pickerItems[0]).toEqual({
      label: '1',
      value: 1,
    });
    expect(pickerItems[pickerItems.length - 1]).toEqual({
      label: '5',
      value: 5,
    });
  });
});
