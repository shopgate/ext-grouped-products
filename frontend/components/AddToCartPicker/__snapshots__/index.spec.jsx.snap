// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddToCartPicker /> should render the component as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AddToCartPicker)
    productId="1337"
    stock={
      Object {
        "ignoreQuantity": false,
        "info": "Info",
        "maxOrderQuantity": 1000,
        "minOrderQuantity": 1,
        "orderable": true,
        "quantity": 4,
      }
    }
  >
    <AddToCartPicker
      addProductToCart={[Function]}
      clickDelay={150}
      maxEntries={5}
      productId="1337"
      stock={
        Object {
          "ignoreQuantity": false,
          "info": "Info",
          "maxOrderQuantity": 1000,
          "minOrderQuantity": 1,
          "orderable": true,
          "quantity": 4,
        }
      }
    >
      <Picker
        buttonComponent={[Function]}
        buttonProps={
          Object {
            "addedQuantity": 0,
            "className": "css-mbcx74",
            "handleAddToCart": [Function],
            "isDisabled": false,
            "isLoading": false,
            "isOrderable": true,
            "noShadow": true,
          }
        }
        className="css-1ej66kr"
        disabled={false}
        duration={300}
        isOpen={false}
        items={
          Array [
            Object {
              "label": "1",
              "value": 1,
            },
            Object {
              "label": "2",
              "value": 2,
            },
            Object {
              "label": "3",
              "value": 3,
            },
            Object {
              "label": "4",
              "value": 4,
            },
          ]
        }
        label=""
        listComponent={[Function]}
        modalComponent={[Function]}
        onChange={[Function]}
        onClose={[Function]}
        placeholder="Pick ..."
        value={null}
      >
        <div
          className="css-1ej66kr"
        >
          <PickerAddToCartButton
            addedQuantity={0}
            buttonSize={40}
            className="css-mbcx74"
            disabled={false}
            handleAddToCart={[Function]}
            hasLoading={false}
            iconSize={20}
            isDisabled={false}
            isLoading={false}
            isOrderable={true}
            label="Pick ..."
            noShadow={true}
            openList={[Function]}
            value=""
          >
            <button
              className="css-mbcx74 css-zby7a7 css-31obrv"
              data-test-id="AddToCartButton"
              onClick={[Function]}
            >
              <div
                className="css-rukjns"
                style={
                  Object {
                    "opacity": 1,
                    "transform": "translate3d(0, 300%, 0)",
                  }
                }
              >
                <Tick>
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"/>"
                    size="inherit"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "inherit",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Tick>
              </div>
              <div
                className="css-rukjns"
                onAnimationEnd={[Function]}
                style={
                  Object {
                    "opacity": 1,
                    "transform": "translate3d(0, -50%, 0)",
                  }
                }
              >
                <CartPlus>
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M0 0h24v24H0zm18.31 6l-2.76 5z\\" fill=\\"none\\"/><path d=\\"M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z\\"/>"
                    size="inherit"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M0 0h24v24H0zm18.31 6l-2.76 5z\\" fill=\\"none\\"/><path d=\\"M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "inherit",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </CartPlus>
              </div>
            </button>
          </PickerAddToCartButton>
          <Component
            isOpened={true}
            onClose={[Function]}
          >
            <Component
              duration={300}
              isOpen={false}
              onClose={[Function]}
            >
              <Component
                duration={300}
                isOpen={false}
                onClose={[Function]}
                title="grouped_products.add_to_cart_picker.headline"
              >
                <Component
                  items={
                    Array [
                      Object {
                        "disabled": false,
                        "label": "1",
                        "value": 1,
                      },
                      Object {
                        "disabled": false,
                        "label": "2",
                        "value": 2,
                      },
                      Object {
                        "disabled": false,
                        "label": "3",
                        "value": 3,
                      },
                      Object {
                        "disabled": false,
                        "label": "4",
                        "value": 4,
                      },
                    ]
                  }
                  onSelect={[Function]}
                  selectedIndex={-1}
                >
                  <List>
                    <List
                      className=""
                    >
                      <ul
                        className="css-gkw1j5 "
                      >
                        <ListItem
                          className="css-1brrprg css-1ym3l2n"
                          isSelected={false}
                          key="child-0/.$1"
                        >
                          <li
                            className="css-1p5q8vq css-1brrprg css-1ym3l2n css-16gj3o1"
                          >
                            <div
                              className="css-1nnz83b"
                            >
                              <Item
                                className={null}
                                image={null}
                                isDisabled={false}
                                isSelected={false}
                                key="1"
                                link={null}
                                onClick={[Function]}
                                rightComponent={null}
                                testId={null}
                                title="1"
                              >
                                <div
                                  aria-hidden={true}
                                  data-test-id={null}
                                  onClick={[Function]}
                                >
                                  <Glow
                                    className={null}
                                    color="#f7f7f7"
                                    styles={
                                      Object {
                                        "container": null,
                                        "glow": null,
                                      }
                                    }
                                  >
                                    <div
                                      aria-hidden={true}
                                      className="css-13azwyo"
                                      onTouchTap={[Function]}
                                      style={null}
                                    >
                                      <div
                                        className="css-1bmd5dz"
                                        style={Object {}}
                                      />
                                      <Grid
                                        className="css-1qcr7hz"
                                        component="div"
                                        wrap={false}
                                      >
                                        <div
                                          className="css-1qcr7hz css-h7r55a"
                                        >
                                          <div
                                            className="css-jwxytc"
                                          />
                                          <GridItem
                                            className="css-10fxwvh"
                                            component="div"
                                            grow={1}
                                            shrink={1}
                                          >
                                            <div
                                              className="css-10fxwvh css-1gs0ko2"
                                            >
                                              1
                                            </div>
                                          </GridItem>
                                        </div>
                                      </Grid>
                                    </div>
                                  </Glow>
                                </div>
                              </Item>
                            </div>
                          </li>
                        </ListItem>
                        <ListItem
                          className="css-1brrprg css-1ym3l2n"
                          isSelected={false}
                          key="child-1/.$2"
                        >
                          <li
                            className="css-1p5q8vq css-1brrprg css-1ym3l2n css-16gj3o1"
                          >
                            <div
                              className="css-1nnz83b"
                            >
                              <Item
                                className={null}
                                image={null}
                                isDisabled={false}
                                isSelected={false}
                                key="2"
                                link={null}
                                onClick={[Function]}
                                rightComponent={null}
                                testId={null}
                                title="2"
                              >
                                <div
                                  aria-hidden={true}
                                  data-test-id={null}
                                  onClick={[Function]}
                                >
                                  <Glow
                                    className={null}
                                    color="#f7f7f7"
                                    styles={
                                      Object {
                                        "container": null,
                                        "glow": null,
                                      }
                                    }
                                  >
                                    <div
                                      aria-hidden={true}
                                      className="css-13azwyo"
                                      onTouchTap={[Function]}
                                      style={null}
                                    >
                                      <div
                                        className="css-1bmd5dz"
                                        style={Object {}}
                                      />
                                      <Grid
                                        className="css-1qcr7hz"
                                        component="div"
                                        wrap={false}
                                      >
                                        <div
                                          className="css-1qcr7hz css-h7r55a"
                                        >
                                          <div
                                            className="css-jwxytc"
                                          />
                                          <GridItem
                                            className="css-10fxwvh"
                                            component="div"
                                            grow={1}
                                            shrink={1}
                                          >
                                            <div
                                              className="css-10fxwvh css-1gs0ko2"
                                            >
                                              2
                                            </div>
                                          </GridItem>
                                        </div>
                                      </Grid>
                                    </div>
                                  </Glow>
                                </div>
                              </Item>
                            </div>
                          </li>
                        </ListItem>
                        <ListItem
                          className="css-1brrprg css-1ym3l2n"
                          isSelected={false}
                          key="child-2/.$3"
                        >
                          <li
                            className="css-1p5q8vq css-1brrprg css-1ym3l2n css-16gj3o1"
                          >
                            <div
                              className="css-1nnz83b"
                            >
                              <Item
                                className={null}
                                image={null}
                                isDisabled={false}
                                isSelected={false}
                                key="3"
                                link={null}
                                onClick={[Function]}
                                rightComponent={null}
                                testId={null}
                                title="3"
                              >
                                <div
                                  aria-hidden={true}
                                  data-test-id={null}
                                  onClick={[Function]}
                                >
                                  <Glow
                                    className={null}
                                    color="#f7f7f7"
                                    styles={
                                      Object {
                                        "container": null,
                                        "glow": null,
                                      }
                                    }
                                  >
                                    <div
                                      aria-hidden={true}
                                      className="css-13azwyo"
                                      onTouchTap={[Function]}
                                      style={null}
                                    >
                                      <div
                                        className="css-1bmd5dz"
                                        style={Object {}}
                                      />
                                      <Grid
                                        className="css-1qcr7hz"
                                        component="div"
                                        wrap={false}
                                      >
                                        <div
                                          className="css-1qcr7hz css-h7r55a"
                                        >
                                          <div
                                            className="css-jwxytc"
                                          />
                                          <GridItem
                                            className="css-10fxwvh"
                                            component="div"
                                            grow={1}
                                            shrink={1}
                                          >
                                            <div
                                              className="css-10fxwvh css-1gs0ko2"
                                            >
                                              3
                                            </div>
                                          </GridItem>
                                        </div>
                                      </Grid>
                                    </div>
                                  </Glow>
                                </div>
                              </Item>
                            </div>
                          </li>
                        </ListItem>
                        <ListItem
                          className="css-1brrprg"
                          isSelected={false}
                          key="child-3/.$4"
                        >
                          <li
                            className="css-1p5q8vq css-1brrprg css-16gj3o1"
                          >
                            <div
                              className="css-1nnz83b"
                            >
                              <Item
                                className={null}
                                image={null}
                                isDisabled={false}
                                isSelected={false}
                                key="4"
                                link={null}
                                onClick={[Function]}
                                rightComponent={null}
                                testId={null}
                                title="4"
                              >
                                <div
                                  aria-hidden={true}
                                  data-test-id={null}
                                  onClick={[Function]}
                                >
                                  <Glow
                                    className={null}
                                    color="#f7f7f7"
                                    styles={
                                      Object {
                                        "container": null,
                                        "glow": null,
                                      }
                                    }
                                  >
                                    <div
                                      aria-hidden={true}
                                      className="css-13azwyo"
                                      onTouchTap={[Function]}
                                      style={null}
                                    >
                                      <div
                                        className="css-1bmd5dz"
                                        style={Object {}}
                                      />
                                      <Grid
                                        className="css-1qcr7hz"
                                        component="div"
                                        wrap={false}
                                      >
                                        <div
                                          className="css-1qcr7hz css-h7r55a"
                                        >
                                          <div
                                            className="css-jwxytc"
                                          />
                                          <GridItem
                                            className="css-10fxwvh"
                                            component="div"
                                            grow={1}
                                            shrink={1}
                                          >
                                            <div
                                              className="css-10fxwvh css-1gs0ko2"
                                            >
                                              4
                                            </div>
                                          </GridItem>
                                        </div>
                                      </Grid>
                                    </div>
                                  </Glow>
                                </div>
                              </Item>
                            </div>
                          </li>
                        </ListItem>
                      </ul>
                    </List>
                  </List>
                </Component>
              </Component>
            </Component>
          </Component>
        </div>
      </Picker>
    </AddToCartPicker>
  </Connect(AddToCartPicker)>
</Provider>
`;
